trustchain-ion/src/attestor.rs

```diff
@@ -168,11 +169,51 @@ impl Issuer for IONAttestor {
     }
 }
 
+#[async_trait]
+impl Holder for IONAttestor {
+    async fn sign_presentation<T: DIDResolver>(
+        &self,
+        presentation: &Presentation,
+        key_id: Option<&str>,
+        resolver: &T,
+        ldp_options: Option<LinkedDataProofOptions>,
+    ) -> Result<Presentation, HolderError> {
+        // If no ldp options passed, use default with ProofPurpose::Authentication.
+        let options = ldp_options.unwrap_or(LinkedDataProofOptions {
+            proof_purpose: Some(ssi::vc::ProofPurpose::Authentication),
+            ..Default::default()
+        });
+
+        // Get the signing key.
+        let signing_key = self.signing_key(key_id)?;
+
+        let mut vp = presentation.clone();
+        // Check holder field is correctly populated
+        match presentation.holder.as_ref() {
+            Some(URI::String(holder)) => {
+                if holder != &self.did {
+                    vp.holder = Some(URI::String(self.did.clone()))
+                }
```

Comment:
Following discussion, update to return an error variant related to incompatible holders.